velox/dwio/dwrf/test/TestColumnReader.cpp

```diff
@@ -727,6 +735,61 @@ TEST_P(TestColumnReader, testByteSkipsWithNulls) {
   EXPECT_TRUE(byteBatch->isNullAt(4));
 }
 
+TEST_P(TestColumnReader, testIntegerRLEv2) {
+  int32_t expects[] = {2030, 2000, 2020, 1000000, 2040, 2050, 2060,
+                       2070, 2080, 2090, 2100,    2110, 2120, 2130,
+                       2140, 2150, 2160, 2170,    2180, 2190};
+  int32_t expects2[] = {2100, 2110, 2120, 2130, 2140};
+  // RLEv2 with PATCHED_BASE
+  const unsigned char buffer[] = {0x8e, 0x13, 0x2b, 0x21, 0x07, 0xd0, 0x1e,
+                                  0x00, 0x14, 0x70, 0x28, 0x32, 0x3c, 0x46,
+                                  0x50, 0x5a, 0x64, 0x6e, 0x78, 0x82, 0x8c,
+                                  0x96, 0xa0, 0xaa, 0xb4, 0xbe, 0xfc, 0xe8};
+  int32_t size = VELOX_ARRAY_SIZE(expects);
+  // set format
+  streams_.setFormat(DwrfFormat::kOrc);
+  // set getEncoding
+  proto::ColumnEncoding directEncoding;
+  proto::ColumnEncoding directv2Encoding;
+  directEncoding.set_kind(proto::ColumnEncoding_Kind_DIRECT);
+  directv2Encoding.set_kind(proto::ColumnEncoding_Kind_DIRECT_V2);
+  EXPECT_CALL(streams_, getEncodingProxy(0))
+      .WillRepeatedly(Return(&directEncoding)); // row type use direct only
+  EXPECT_CALL(streams_, getEncodingProxy(1))
+      .WillRepeatedly(Return(&directv2Encoding));
+  // set getStream
+  EXPECT_CALL(streams_, getStreamProxy(_, proto::Stream_Kind_PRESENT, false))
+      .WillRepeatedly(Return(nullptr));
+  EXPECT_CALL(streams_, getStreamProxy(_, proto::Stream_Kind_ROW_INDEX, false))
+      .WillRepeatedly(Return(nullptr));
+  EXPECT_CALL(streams_, getStreamProxy(1, proto::Stream_Kind_DATA, true))
+      .WillRepeatedly(Return(
+          new SeekableArrayInputStream(buffer, VELOX_ARRAY_SIZE(buffer))));
+  // create the row type
+  auto rowType = HiveTypeParser().parse("struct<myInt:int>");
```

Comment:
Can we have 2 columns, one encoded in RLEv2 and one not.  Then test it with a filter on the first column only, and another test for filter on second column only.

Reply:
Adjusted TestColumnReader::testIntegerRLEv2 to enable both Filter simultaneously, do you think this is OK?